lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -303,6 +309,7 @@ messageTypeTag :: MessageType -> Word8
 messageTypeTag = \case
     MsgTypeInitial -> 0
     MsgTypeTransaction -> 2
+    MsgTypeDelegation -> 3
```

Comment:
`3` is for `OwnerStakeDelegation` which is not what we are looking into here in the end :( 
Instead, we are after `TYPE-STAKE-DELEGATION (= 0x04)`. 

See also `FRAGMENT-TYPE` in https://github.com/input-output-hk/chain-libs/blob/7a7c74359df7b7fbdb9f71a83c11f88100897c06/chain-impl-mockchain/doc/format.md#block-content